extensions/smallrye-metrics/runtime/src/main/java/io/quarkus/smallrye/metrics/runtime/SmallRyeMetricsTemplate.java

```diff
@@ -37,14 +43,82 @@
 
     private static final Logger log = Logger.getLogger("io.quarkus.metrics");
-    private static final String MEMORY_HEAP_USAGE = "memory.heap.usage";
-    private static final String MEMORY_NON_HEAP_USAGE = "memory.nonHeap.usage";
+
+    // threading
     private static final String THREAD_COUNT = "thread.count";
+    private static final String THREAD_DAEMON_COUNT = "thread.daemon.count";
+    private static final String THREAD_MAX_COUNT = "thread.max.count";
+
+    // class loading
+    private static final String CURRENT_LOADED_CLASS_COUNT = "classloader.currentLoadedClass.count";
+    private static final String TOTAL_LOADED_CLASS_COUNT = "classloader.totalLoadedClass.count";
+    private static final String TOTAL_UNLOADED_CLASS_COUNT = "classloader.totalUnloadedClass.count";
+
+    // runtime
+    private static final String JVM_UPTIME = "jvm.uptime";
+
+    // operating system
+    private static final String SYSTEM_LOAD_AVERAGE = "cpu.systemLoadAverage";
+    private static final String CPU_AVAILABLE_PROCESSORS = "cpu.availableProcessors";
+
+    // memory
+    private static final String MEMORY_COMMITTED_NON_HEAP = "memory.committedNonHeap";
+    private static final String MEMORY_COMMITTED_HEAP = "memory.committedHeap";
+    private static final String MEMORY_MAX_HEAP = "memory.maxHeap";
+    private static final String MEMORY_MAX_NON_HEAP = "memory.maxNonHeap";
+    private static final String MEMORY_USED_HEAP = "memory.usedHeap";
+    private static final String MEMORY_USED_NON_HEAP = "memory.usedNonHeap";
+
+    public void registerVendorMetrics(ShutdownContext shutdown) {
+        MetricRegistry registry = MetricRegistries.get(MetricRegistry.Type.VENDOR);
+        List<String> names = new ArrayList<>();
+
+        memoryPoolMetrics(registry, names);
+        vendorSpecificMemoryMetrics(registry, names);
+
+        if (!names.isEmpty()) {
+            shutdown.addShutdownTask(() -> {
+                for (String i : names) {
+                    registry.remove(i);
+                }
+            });
+        }
+    }
 
     public void registerBaseMetrics(ShutdownContext shutdown) {
         MetricRegistry registry = MetricRegistries.get(MetricRegistry.Type.BASE);
-        List<GarbageCollectorMXBean> gcs = ManagementFactory.getGarbageCollectorMXBeans();
         List<String> names = new ArrayList<>();
+
+        garbageCollectionMetrics(registry, names);
+        classLoadingMetrics(registry, names);
+        operatingSystemMetrics(registry, names);
+        threadingMetrics(registry, names);
+        runtimeMetrics(registry, names);
+        baseMemoryMetrics(registry, names);
+
+        if (!names.isEmpty()) {
+            shutdown.addShutdownTask(() -> {
+                for (String i : names) {
+                    registry.remove(i);
+                }
+            });
+        }
+    }
+
+    public void createRegistries(BeanContainer container) {
+        log.info("Creating registries");
```

Comment:
This needs to be dropped to debug

Reply:
Yeah, I spotted this one too. I think we can remove it altogether. It doesn't bring any useful information.